### PR TITLE
Assert `and` and `or` create no new scope

### PR DESCRIPTION
**Pull request by scorphus**
## Assert `and` and `or` create no new scope

This PR suggests the addition of 2 set of tests that assert that variables declared as part of `and` and `or` statements get declared in the expected scope.

There's not issue related to this change. This PR comes as a result of a [brief conversation I had with @zanchey over Gitter](https://gitter.im/fish-shell/fish-shell?at=61d99374526fb77b316c0218).

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst

I believe none of the TODOs above apply.

I hope this is a good first contribution and I'm happy to make any change to it.

### CODE REVIEW

File: tests/checks/set.fish [hunk 1]
```diff
@@ -635,6 +635,36 @@ end
...
+#CHECK: $var7a[1]: |89|
+#CHECK: $var7a[2]: |179|
+true; and begin
+    set -l -a var7b 359 719
+    set --show var7b
+end
+set --show var7b
+#CHECK: $var7b: set in local scope, unexported, with 2 elements
+#CHECK: $var7b[1]: |359|
+#CHECK: $var7b[2]: |719|
```
Comment (krobelus) [inline tests/checks/set.fish:651]: I found this a bit hard to follow.

In the gitter link you summarize this as "and/or do not create a new scope".
This is quite important to understand the point of this test, so it should be in a comment or commit message.

You add have scenarios `true; and set someglobal` and `true; and begin; set -l somelocal; end` but I don't see a test that actually ensures that a and/or don't create a new scope.
That would need something like `true; and set -l local; set -q local`.
I guess a minimal test along those lines could work.

Comment (scorphus) [reply tests/checks/set.fish:651]: Your inputs are helpful, @krobelus. Thanks. I'll soon update this PR.

**Comment by faho** [comment]
Merged, thanks!
